src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -514,18 +514,20 @@ private static boolean startsWithInheritDoc(DetailNode root) {
         boolean found = false;
         final DetailNode[] children = root.getChildren();
 
-        for (int i = 0; !found; i++) {
-            final DetailNode child = children[i];
-            if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG
-                    && child.getChildren()[1].getType() == JavadocTokenTypes.INHERIT_DOC_LITERAL) {
-                found = true;
-            }
-            else if (child.getType() != JavadocTokenTypes.LEADING_ASTERISK
-                    && !CommonUtil.isBlank(child.getText())) {
+        int i = 0;
+        while (i != children.length) {
+            if ((children[i].getType() == JavadocTokenTypes.TEXT
+                    || children[i].getType() == JavadocTokenTypes.HTML_ELEMENT)
```

Comment:
you are not using `i` for anything except for `childern[i]` it is singnal that you need foreach `for(DetailAst child: children)`